kso_utils/db_utils.py

```diff
@@ -177,31 +178,30 @@ def cols_rename_to_schema(
     :param table_name: The name of the table in the database where the data is stored
     :param reverse_lookup: a boolean value to reverse the dict if formatting from schema to csv
     """
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError(f"Expected a pandas Dataframe for df, got {type(df)} instead.")
 
-    # Get the spyfish-specific column names and their correspondent
-    # schema fields
-    if project.Project_name in ["Spyfish_Aotearoa", "Spyfish_BOPRC"]:
-        from kso_utils.spyfish_utils import get_spyfish_col_names
-
-        col_names_lookup = get_spyfish_col_names(table_name)
-
-    # Get the koster-specific column names and their correspondent
-    # schema fields
-    if project.Project_name == "Koster_Seafloor_Obs":
-        from kso_utils.koster_utils import get_koster_col_names
-
-        col_names_lookup = get_koster_col_names(table_name)
-
-    # Rename project-specific columns using the dictionary
-    if "col_names_lookup" in locals():
-        if reverse_lookup:
-            # Reverse the dictionaries if formatting from schema to csv
-            col_names_lookup = dict(
-                zip(col_names_lookup.values(), col_names_lookup.keys())
-            )
-
-        df = df.rename(columns=col_names_lookup)
-
+    # Check if there is a function to rename the columns
+    try:
+        mod = import_module(project.utils_path)
+        func_get_col_names = getattr(mod, "get_col_names")
+        col_names_lookup = func_get_col_names(table_name)
+        logging.info(f"{project.Project_name} has colums to rename, these are successfully retrieved.")
+    except (ModuleNotFoundError, AttributeError) as e:
+        logging.info(f"{project.Project_name} does not contain any columns that need renaming. So the original column names are kept.")
```

Comment:
One thing that Python linters will complain about, is not to use f-strings in logging functions. Instead use:

`logging.info("%s does not contain any columns that need renaming. So the original column names are kept.", project.Project_name)`

 The reason is that logging won't replace the `%s` and create a new string unless the string *has* to be logged. If the string doesn't, because the log level is higher, it ignores it. By using an f-string, Python automatically replaces it, regardless of whether it'll be used or not